src/Microsoft.VisualStudio.SolutionPersistence/Model/SolutionModel.cs

```diff
@@ -271,11 +279,16 @@ public void AddBuildType(string buildType)
 
         ValidateName(buildType.AsSpan());
 
-        if (!this.solutionBuildTypes.Contains(buildType, StringComparer.OrdinalIgnoreCase))
+        foreach (var type in this.solutionBuildTypes)
         {
-            buildType = this.StringTable.GetString(buildType);
-            this.solutionBuildTypes.Add(buildType);
+            if (string.Equals(type, buildType, StringComparison.OrdinalIgnoreCase))
```

Comment:
This isn't the same logic. The original was a `Contains`, but this new code is an `Equals`.